kasparovsyncd/sync/sync.go

```diff
@@ -405,38 +427,48 @@ func handleBlockAddedMsg(client *jsonrpc.Client, blockAdded *jsonrpc.BlockAddedM
 		return nil
 	}
 
-	err = fetchAndAddBlock(client, dbTx, blockHash)
+	addedBlockHashes, err := fetchAndAddBlock(client, dbTx, blockHash)
 	if err != nil {
 		return err
 	}
-	return dbTx.Commit()
+
+	err = dbTx.Commit()
+	if err != nil {
+		return err
+	}
+
+	for _, hash := range addedBlockHashes {
+		err := mqtt.PublishBlockAddedNotifications(hash)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-func fetchAndAddBlock(client *jsonrpc.Client, dbTx *dbaccess.TxContext, blockHash *daghash.Hash) error {
+func fetchAndAddBlock(client *jsonrpc.Client, dbTx *dbaccess.TxContext, blockHash *daghash.Hash) (addedBlockHashes []string, err error) {
 	block, err := fetchBlock(client, blockHash)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	missingAncestors, err := fetchMissingAncestors(client, dbTx, block, nil)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	blocks := append([]*rawAndVerboseBlock{block}, missingAncestors...)
 	err = bulkInsertBlocksData(client, dbTx, blocks)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	for _, block := range blocks {
-		err := mqtt.PublishTransactionsNotifications(block.Verbose.RawTx)
-		if err != nil {
-			return err
-		}
+	addedBlockHashes = make([]string, len(blocks))
```

Comment:
suggestion: maybe you could return the `blocks` array itself, rather than creating the sting array, and then loop over that array again to send the notifications?

Reply:
oh i see that use it in `fetchAndAddMissingAddedChainBlocks` as well, so it makes sense to keep it this way. please disregard the above comment .